src/checks/piiDetection.ts

```diff
@@ -0,0 +1,258 @@
+import { Readable } from 'stream'
+
+import * as core from '@actions/core'
+import { WebhookEventMap } from '@octokit/webhooks-types'
+import { Endpoints } from '@octokit/types'
+import CsvReadableStream from 'csv-reader'
+
+import { github } from '../infrastructure/github'
+
+import Check from './check'
+
+type FileData =
+  Endpoints['GET /repos/{owner}/{repo}/commits/{ref}']['response']['data']['files']
+
+export type PiiDataTypeName =
+  | 'us-phone-number'
+  | 'email'
+  | 'ssn'
+  | 'credit-card-number'
+
+export type Prediction = {
+  detected: boolean
+  dataType?: PiiDataTypeName[]
+}
+
+export type piiDataType = {
+  type: PiiDataTypeName
+  regexp: RegExp
+}
+
+export type PredictionResults = {
+  file: string
+  prediction: Prediction
+}[]
+
+const piiData: piiDataType[] = [
+  {
+    type: 'us-phone-number',
+    regexp: new RegExp(
+      /^\+?(\d+)?[ .-]?\(?(\d{3})\)?[ .-]?(\d{3})[ .-]?(\d{4})$/i
+    ), // source https://gist.github.com/charles-rumley/e13b314662a203e5172a298bc66544b3
+  },
+  {
+    type: 'email',
+    regexp: new RegExp(/^[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+$/), // source https://ihateregex.io/expr/email
+  },
+  {
+    type: 'ssn',
+    regexp: new RegExp(
+      /^(?!0{3})(?!6{3})[0-8]\d{2}-(?!0{2})\d{2}-(?!0{4})\d{4}$/
+    ), // source https://ihateregex.io/expr/ssn/
+  },
+  {
+    type: 'credit-card-number',
+    regexp: new RegExp(
+      /^(^4[0-9]{12}(?:[0-9]{3})?$)|(^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$)|(3[47][0-9]{13})|(^3(?:0[0-5]|[68][0-9])[0-9]{11}$)|(^6(?:011|5[0-9]{2})[0-9]{12}$)|(^(?:2131|1800|35\d{3})\d{11}$)$/
+    ), // source https://ihateregex.io/expr/credit-card/
+  },
+]
+
+const FileExtensionsToScan = ['.csv']
+export const Ignorefile = '.piidetectionignore'
+export const CsvDetectionThreshold = 0.7 // At least 70% of the CSV file lines must contain one of the detected data types
+const NotionPage =
+  'https://www.notion.so/typeform/PII-Detection-Check-11658ceb312c49a69681d2507e750748'
+
+const piiDetection: Check = {
+  name: 'pii-detection',
+  async run(): Promise<boolean> {
+    let filesData
+    if (github.context.eventName === 'push') {
+      filesData = await downloadFilesDataFromPush()
+    } else if (github.context.eventName === 'pull_request') {
+      filesData = await downloadFilesDataFromPullRequest()
+    } else {
+      core.info(
+        'PII detection will only run on "push" and "pull_request" events. Skipping...'
+      )
+      return true
+    }
```

Comment:
to avoid some conditionals we do:


```suggestion
   if ( !['push', 'pull_request'].includes(github.context.eventName) ) {
   core.info(
        'PII detection will only run on "push" and "pull_request" events. Skipping...'
      )
      return true
}

const filesData = github.context.eventName === 'push'
       ? await downloadFilesDataFromPush()
       : await downloadFilesDataFromPullRequest()
```